jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/ArrayByteBufferPool.java

```diff
@@ -520,18 +533,85 @@ public String toString()
                     inUse++;
             }
 
-            return String.format("%s{capacity=%d,inuse=%d(%d%%)}",
+            return String.format("%s{capacity=%d,in-use=%d/%d,pooled/acquires=%d/%d,non-pooled/evicts/removes/releases=%d/%d/%d/%d}",
                 super.toString(),
-                _capacity,
+                getCapacity(),
                 inUse,
-                entries > 0 ? (inUse * 100) / entries : 0);
+                entries,
+                _pooled.get(),
+                _acquires.get(),
+                _nonPooled.get(),
+                _evicts.get(),
+                _removes.get(),
+                _releases.get()
+            );
+        }
+
+        private class CompoundBucket extends CompoundPool<RetainableByteBuffer>
+        {
+            private CompoundBucket(ConcurrentBucket concurrentBucket, QueuedBucket queuedBucket)
+            {
+                super(concurrentBucket, queuedBucket);
+            }
+
+            private Pool.Entry<RetainableByteBuffer> evict()
```

Comment:
This method is pretty simple and only called from one place. It could be inlined which would save having this internal class.